libs/blocks/youtube/youtube.js

```diff
@@ -1,36 +1,111 @@
-import { createIntersectionObserver, isInTextNode } from '../../utils/utils.js';
+import { createIntersectionObserver, createTag, isInTextNode } from '../../utils/utils.js';
 
-export default function init(a) {
+class LiteYTEmbed extends HTMLElement {
+  connectedCallback() {
+    this.videoId = this.getAttribute('videoid');
+    const playBtnEl = createTag('button', { type: 'button', class: 'lty-playbtn' });
+    this.append(playBtnEl);
+    this.playLabel = this.getAttribute('playlabel') || 'Play';
+    this.style.backgroundImage = `url("https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg")`;
+    this.style.backgroundSize = 'cover';
+    this.style.backgroundPosition = 'center';
+    const playBtnLabelEl = createTag('span', { class: 'lyt-visually-hidden' });
+    playBtnLabelEl.textContent = this.playLabel;
+    playBtnEl.append(playBtnLabelEl);
+    this.addEventListener('pointerover', LiteYTEmbed.warmConnections, { once: true });
+    this.addEventListener('click', this.addIframe);
+    this.needsYTApiForAutoplay = navigator.vendor.includes('Apple') || navigator.userAgent.includes('Mobi');
+  }
+
+  static addPrefetch(kind, url, as) {
+    const linkEl = createTag('link', { rel: kind, href: url });
+    if (as) linkEl.as = as;
+    document.head.append(linkEl);
+  }
+
+  static warmConnections() {
+    if (LiteYTEmbed.preconnected) return;
+    LiteYTEmbed.addPrefetch('preconnect', 'https://www.youtube-nocookie.com');
+    LiteYTEmbed.addPrefetch('preconnect', 'https://www.google.com');
+    LiteYTEmbed.addPrefetch('preconnect', 'https://googleads.g.doubleclick.net');
+    LiteYTEmbed.addPrefetch('preconnect', 'https://static.doubleclick.net');
+    LiteYTEmbed.preconnected = true;
```

Comment:
OCD nit: for Vimeo you declare this before adding the preconnects. None is better or worse, but consistency is always nicer on the eye 🙂

Reply:
Now that you've pointed it out I cannot unsee it and my OCD kicked in 😂 I've made the change 👍